docs/application/native/guides/text-input/multi-assistant.md

```diff
@@ -0,0 +1,444 @@
+
+To use functionalities of the Multi-assistant framework:
+
+1. Set up Multi-assistant client library and [register callbacks](#callback).
+
+   The initialization allows the Multi-assistant framework to distinguish your application from any other applications also using the framework. The registered callbacks allow you to receive notifications about changes in the client state, service state and active state, and enables retrieving user utterance audio data from the framework.
+
+2. Prepare the Multi-assistant client library.
+
+   The preparation connects the client application to the Multi-assistant service to start exchanging messages with Mutli-assistant framework and wakeup engines.
+
+3. Retrieve audio data when activated.
+
+   If the registered active state changed event callback function is called with the ACTIVE parameter, then you must ask for the user utterance data so that the query could be processed by your voice assistant.
+
+4. Send the result.
+
+   While you are processing the user utterance, it is recomended to update the partially converted ASR result to the Multi-assistant framework so that it can display the visual feedback to the user.
+
+   When the processing work is finished, you can send the final result to the Multi-assistant framework to display the result, and also send recognition result to report whether the audio data processing has succeeded or not.
+
+5. When no longer needed, unprepare and deinitialize the Multi-assistant client library.
+
+   You must disconnect from the Multi-assistant service and deinitialize the client library using `ma_unprepare()` and `ma_deinitialize()`.
+
+
+## Prerequisites
+
+To enable your application to use the Multi-assistant functionality:
+
+1. To use the functions and data types of the Multi-assistant API (in [mobile](../../api/mobile/latest/group__CAPI__UIX__MULTI__ASSISTANT__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__UIX__MULTI__ASSISTANT__MODULE.html) applications), include the `<multi_assistant.h>` header file in your application:
+
+   ```c
+   #include <multi_assistant.h>
+   ```
+
+2. Initialize the Multi-assistant client library with `ma_initialize()`:
+
+   ```c
+   void
+   initialize_multi_assistant()
+   {
+       int ret;
+       ret = ma_initialize();
+       if (MA_ERROR_NONE != result)
+           /* Error handling */
+   }
+   ```
+
+   If the function call is successful, the client state changes to `MA_STATE_INITIALIZED`.
+
+   > [!NOTE]
+   > The Mutli-assistant client feature is not thread-safe and depends on the Ecore main loop. Implement Multi-assistant within the Ecore main loop and do not use it in a thread.
+
+3. Prepare Multi-assistant with `ma_prepare()`, which connects the client application to the Multi-assistant service. The Multi-assistant service records user utterance and manages wakeup engines to detect the wake word from the audio data.
```

Comment:
Please use colon (:) instead of a period at the end of the sentence as it is followed by code.

3. Prepare Multi-assistant with `ma_prepare()`, which connects the client application to the Multi-assistant service. The Multi-assistant service records user utterance and manages wakeup engines to detect the wake word from the audio data:

Reply:
Suggested changes uploaded : 2c1a24a1e02be6d3d3b9fd1b0a1fcfdae4c07534